lib/msal-common/src/authority/Authority.ts

```diff
@@ -257,6 +261,19 @@ export class Authority {
 
         metadata = await this.getEndpointMetadataFromNetwork();
         if (metadata) {
+            // If the user prefers to use an azure region replace the global endpoints with regional information.
+            if (this.authorityOptions.azureRegionConfiguration?.preferredAzureRegion) {
+                const autodetectedRegionName = await this.regionDiscovery.detectRegion(this.authorityOptions.azureRegionConfiguration.environmentRegion);
+
+                const azureRegion = this.authorityOptions.azureRegionConfiguration.preferredAzureRegion === Constants.AZURE_REGION_AUTO_DISCOVER_FLAG 
+                    ? autodetectedRegionName 
+                    : this.authorityOptions.azureRegionConfiguration.preferredAzureRegion;
+
+                if (azureRegion) {
+                    metadata = Authority.replaceWithRegionalInformation(metadata, azureRegion);
+                }
+            }
+
```

Comment:
So this code fragment ensures we do the instance_discovery with global authority and then add regional_data right? Can we please add a test case for it?